microceph/ceph/osd.go

```diff
@@ -68,12 +71,141 @@ func nextOSD(s *state.State) (int64, error) {
 	}
 }
 
+// setupEncryptedOSD sets up an encrypted OSD on the given disk.
+//
+// Takes a path to the disk device as well as the osd data path and the osd id.
+// Returns the path to the encrypted device and an error if any.
+func setupEncryptedOSD(devicePath string, osdDataPath string, osdID int64) (string, error) {
+	if err := os.Symlink(devicePath, filepath.Join(osdDataPath, "unencrypted")); err != nil {
+		return "", fmt.Errorf("Failed to add unencrypted block symlink: %w", err)
+	}
+	// Create a key for the encrypted device
```

Comment:
Code style nit, put you'll want a line break before that comment.
Same for the chunks below.